browser/net/brave_site_hacks_network_delegate_helper.cc

```diff
@@ -26,75 +29,44 @@ namespace brave {
-           "_openstat",
-           // https://github.com/brave/brave-browser/issues/11817
-           "vero_conv", "vero_id",
-           // https://github.com/brave/brave-browser/issues/13647
-           "wickedid",
-           // https://github.com/brave/brave-browser/issues/11578
-           "yclid",
-           // https://github.com/brave/brave-browser/issues/8975
-           "__s",
-           // https://github.com/brave/brave-browser/issues/17451
-           "rb_clickid",
-           // https://github.com/brave/brave-browser/issues/17452
-           "s_cid",
-           // https://github.com/brave/brave-browser/issues/17507
-           "ml_subscriber", "ml_subscriber_hash",
-           // https://github.com/brave/brave-browser/issues/18020
-           "twclid",
-           // https://github.com/brave/brave-browser/issues/18758
-           "gbraid", "wbraid",
-           // https://github.com/brave/brave-browser/issues/9019
-           "_hsenc", "__hssc", "__hstc", "__hsfp", "hsCtaTracking",
-           // https://github.com/brave/brave-browser/issues/22082
-           "oft_id", "oft_k", "oft_lk", "oft_d", "oft_c", "oft_ck", "oft_ids",
-           "oft_sk",
-           // https://github.com/brave/brave-browser/issues/11580
-           "igshid"}),
-      "|"));
-  return *trackers;
-}
-
-// From src/components/autofill/content/renderer/page_passwords_analyser.cc
-// and password_form_conversion_utils.cc:
-#define DECLARE_LAZY_MATCHER(NAME, PATTERN)                                   \
-  struct LabelPatternLazyInstanceTraits_##NAME                                \
-      : public base::internal::DestructorAtExitLazyInstanceTraits<re2::RE2> { \
-    static re2::RE2* New(void* instance) {                                    \
-      re2::RE2::Options options;                                              \
-      options.set_case_sensitive(false);                                      \
-      re2::RE2* matcher = new (instance) re2::RE2(PATTERN, options);          \
-      DCHECK(matcher->ok());                                                  \
-      return matcher;                                                         \
-    }                                                                         \
-  };                                                                          \
-  base::LazyInstance<re2::RE2, LabelPatternLazyInstanceTraits_##NAME> NAME =  \
-      LAZY_INSTANCE_INITIALIZER
-
-// e.g. "?fbclid=1234"
-DECLARE_LAZY_MATCHER(tracker_only_matcher,
-                     "^(" + GetQueryStringTrackers() + ")=[^&]+$");
-
-// e.g. "?fbclid=1234&foo=1"
-DECLARE_LAZY_MATCHER(tracker_first_matcher,
-                     "^(" + GetQueryStringTrackers() + ")=[^&]+&");
-
-// e.g. "?foo=1&fbclid=1234" or "?foo=1&fbclid=1234&bar=2"
-DECLARE_LAZY_MATCHER(tracker_appended_matcher,
-                     "&(" + GetQueryStringTrackers() + ")=[^&]+");
-
-#undef DECLARE_LAZY_MATCHER
+const auto simple_trackers = std::set<std::string>(
```

Comment:
please take a look at the code convention ([global objects with non-trivial destructors are banned](https://google.github.io/styleguide/cppguide.html#Static_and_Global_Variables)) and `base::NoDestructor` concept, it has been here on purpose

I think in this particular case you can use `base::MakeFixedFlatSet/Map` + `base::StringPiece` (//base/containers/README.md is also a good read!)

Reply:
Thanks @iefremov  -- you were right, I was able to use `base::MakeFixedFlatSet/Map` + `base::StringPiece`